components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+                            endpoint = gatewayEndpoint.split(",")[0];
+                        } else {
+                            endpoint = gatewayEndpoint;
+                        }
+                    } else {
+                        for (String environmentName : publishedEnvironments) {
+                            // find environment that has hybrid type
+                            if (APIConstants.GATEWAY_ENV_TYPE_HYBRID
+                                    .equals(environments.get(environmentName).getType())) {
+                                environment = environments.get(environmentName);
+                                break;
+                            }
+                        }
+                        //if not having any hybrid environment give 1st environment in api published list
+                        if (environment == null) {
+                            environment = environments.get(publishedEnvironments.toArray()[0]);
+                        }
+                        String gatewayEndpoint = environment.getApiGatewayEndpoint();
+                        if (gatewayEndpoint != null && gatewayEndpoint.contains(",")) {
+                            endpoint = gatewayEndpoint.split(",")[0];
+                        } else {
+                            endpoint = gatewayEndpoint;
+                        }
+                    }
+                }
+                //String apiPath = APIUtil.getAPIPath(apiIdentifier);
+                if (endpoint.endsWith(RegistryConstants.PATH_SEPARATOR)) {
+                    endpoint.substring(0, endpoint.length() - 1);
+                }
+                // We do not need the version in the base path since with the context version strategy, the version is
+                // embedded in the context
+                String basePath = endpoint + api.getContext();
+                resourceConfig.put("basePath", basePath);
+
+                JSONArray resource_configs = (JSONArray) resourceConfig.get("apis");
+
+                //Iterating each Sub resourcePath config
+                int subResCount = 0;
+                while (subResCount < resource_configs.size()) {
+                    JSONObject subResource = (JSONObject) resource_configs.get(subResCount);
+                    String uriTempVal = (String) subResource.get("path");
+                    uriTempVal = uriTempVal.startsWith("/") ? uriTempVal : ("/" + uriTempVal);
+
+                    JSONArray operations = (JSONArray) subResource.get("operations");
+                    //Iterating each operation config
+                    for (Object operation1 : operations) {
+                        JSONObject operation = (JSONObject) operation1;
+                        String httpVerb = (String) operation.get("method");
+                        /* Right Now PATCH is not supported. Need to remove this check when PATCH is supported*/
+                        if (!"PATCH".equals(httpVerb)) {
```

Comment:
Generally it's better to move strings to constant class and refer from there which help to maintain the changes in one place.